lib/main.js

```diff
@@ -0,0 +1,101 @@
+'use babel';
+
+// eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
+import { CompositeDisposable } from 'atom';
+
+import * as helpers from 'atom-linter';
+import escapeHtml from 'escape-html';
+import { dirname } from 'path';
+
+// Some internal variables
+let executablePath;
+let globalHamlLintYmlFile;
+
+const regex = /.+?:(\d+) \[(W|E)\] (\w+): (.+)/g;
+const urlBase = 'https://github.com/brigade/haml-lint/blob/master/lib/haml_lint/linter/README.md';
+
+export default {
+  activate() {
+    require('atom-package-deps').install('linter-haml');
+
+    this.subscriptions = new CompositeDisposable();
+    this.subscriptions.add(
+      atom.config.observe('linter-haml.hamlLintExecutablePath', (value) => {
+        executablePath = value;
+      })
+    );
+    this.subscriptions.add(
+      atom.config.observe('linter-haml.globalHamlLintYmlFile', (value) => {
+        globalHamlLintYmlFile = value;
+      })
+    );
+  },
+
+  deactivate() {
+    this.subscriptions.dispose();
+  },
+
+  provideLinter() {
+    return {
+      name: 'HAML-Lint',
+      grammarScopes: ['text.haml'],
+      scope: 'file',
+      lintOnFly: false,
+      lint: (textEditor) => {
+        const filePath = textEditor.getPath();
+        const text = textEditor.getText();
+
+        const options = {
+          cwd: dirname(filePath),
+          ignoreExitCode: true,
+        };
+        const parameters = [];
+
+        // Specify any configuration file
+        const hamlLintYmlPath = helpers.find(options.cwd, '.haml-lint.yml');
+        if (hamlLintYmlPath) {
+          parameters.push('--config', hamlLintYmlPath);
+        } else if (globalHamlLintYmlFile !== '') {
+          parameters.push('--config', globalHamlLintYmlFile);
+        }
+
+        // Add the file to be linted
+        parameters.push(filePath);
+
+        return helpers.exec(executablePath, parameters, options).then((output) => {
+          if (textEditor.getText() !== text) {
+            const warnMessage = 'File contents changed since the lint was triggered';
+            const warnOptions = {
+              description: 'The file was modified since the request was sent ' +
+                'to `linter-haml` to check it. Since any results would no longer ' +
+                'be valid, they are not being updated. Please save the file again ' +
+                'to update the results.',
+              dismissable: true,
+            };
+            atom.notifications.addWarning(warnMessage, warnOptions);
+            // The text has changed since the lint was triggered, tell Linter not to update
+            return null;
+          }
+
+          const messages = [];
+          let match = regex.exec(output);
```

Comment:
I personally don't think it's wrong. Besides the code style guide was not given to us by God so we are free to object and change it to something that helps us keep the codebase clean and unduplicated